compiler/rustc_resolve/src/ident.rs

```diff
@@ -1155,7 +1153,7 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             }
             Res::Def(DefKind::TyParam, _) | Res::SelfTyParam { .. } | Res::SelfTyAlias { .. } => {
                 for rib in ribs {
-                    let has_generic_params: HasGenericParams = match rib.kind {
+                    let (has_generic_params, defkind) = match rib.kind {
```

Comment:
There are 544 usages of `def_kind` in the compiler and 8 usages of `defkind`. Can you rename it to `def_kind`?